forge-gui/res/cardsfolder/upcoming/heap_gate.txt

```diff
@@ -0,0 +1,7 @@
+Name:Heap Gate
+ManaCost:no cost
+Types:Land Gate
+A:AB$ Mana | Cost$ T | Produced$ C | Amount$ 1 | SpellDescription$ Add {C}.
+A:AB$ Mana | Cost$ 1 T | Produced$ Any | SpellDescription$ Add one mana of any color.
+A:AB$ Token | Cost$ 1 T tapXType<1/Gate> | TokenScript$ c_a_treasure_sac | TokenAmount$ 1
+Oracle:{T}: Add {C}.\n{1}, {T}: Add one mana of any color.\n{1}, {T}, Tap an untapped Gate you control: Create a Treasure token. (It's an artifact with "{T}, Sacrifice this artifact: Add one mana of any color.")
```

Comment:
add DeckHas for treasure tokens (`DeckHas:Ability$Sacrifice|Token & Type$Treasure|Artifact`)
add DeckHints for Gate as in other cards since this also likes to have other Gates